lib/client/ClientRenderingWorkflow.js

```diff
@@ -89,20 +88,12 @@ var ClientRenderingWorkflow = {
                         layoutDelegate.replayInstructions();
                         layoutDelegate.stopRecording();
                     }
-
-                    return view;
                 };
             }
 
-            function enterDOM(view) {
-                if (isFirstPageLoad || !layout.isSameAs(currentLayout)) {
-                    layout.enterDOM();
-                }
+            function finishRoute(requestId) {
+                layout.enterDOM();
```

Comment:
this call doesn't work here exactly if behavior needs to be the same. if there is an error in enterDOM, this promise chain would swallow it rather than show error page